ethpm/package.py

```diff
@@ -58,6 +61,8 @@ def set_default_w3(self, w3: Web3) -> None:
         """
         Set the default Web3 instance.
         """
+        if "deployments" in self.__dict__:
+            del self.deployments
```

Comment:
I have an alternate suggestion that I'm not sure is better.  **If** we stick with this one, this deserves a comment to explain why, and probably another comment in the actual `deployments` property function which explains that the cache gets busted via this mechanism.

My idea is to cache the deployments using a cache key based on the web3 instance itself.  That way, when a new web3 instance gets added, the cache is automatically busted.

```python
@property
def deployments(self):
    should_bust_cache = (
        self.deployments is None or
        self.deployments[0] != id(self.w3)
    )
    if should_bust_cache:
        deps = ...  # construct the deps
        self.deployments = (id(w3), deps)
    return self.deployments[1]
```

This code is pretty gross but hopefully it demonstrates the idea.

Reply:
Tinkered around with this idea, but running into an infinite recursion loop (`should_bust_cache` looks up the deployments property, calling the method and `should_bust_cache` again . . .) Know any trick to avoid this?

It would be nice to have a simpler cache strategy (so we can get rid of the `cached_property` util) but, I'm leaning towards leaving it as is - and adding some comments. 

Also occurred to me that we probably want to bust the `build_dependencies` cached property as well upon switching chains.